app/controllers/spree/admin/reports/enterprise_fee_summary_report_controller.rb

```diff
@@ -0,0 +1,64 @@
+require "open_food_network/reports"
+require "order_management/reports/enterprise_fee_summary/parameters"
+require "order_management/reports/enterprise_fee_summary/permissions"
+require "order_management/reports/enterprise_fee_summary/authorizer"
+require "order_management/reports/enterprise_fee_summary/report_service"
+require "order_management/reports/enterprise_fee_summary/renderers/csv_renderer"
+require "order_management/reports/enterprise_fee_summary/renderers/html_renderer"
+
+module Spree
+  module Admin
+    module Reports
+      class EnterpriseFeeSummaryReportController < BaseController
+        def index
+          @report_parameters = OrderManagement::Reports::EnterpriseFeeSummary::Parameters.new(params[:report] || {})
+          @permissions = OrderManagement::Reports::EnterpriseFeeSummary::Permissions.new(spree_current_user)
+          @authorizer = OrderManagement::Reports::EnterpriseFeeSummary::Authorizer.new(@report_parameters, @permissions)
+
+          return render_report_form if params[:report].blank?
+          return respond_to_invalid_parameters unless @report_parameters.valid?
+
+          @authorizer.authorize!
+
+          @report = OrderManagement::Reports::EnterpriseFeeSummary::ReportService.new(@permissions, @report_parameters, report_renderer_klass)
+          render_report
+        rescue OpenFoodNetwork::Reports::Authorizer::ParameterNotAllowedError => e
+          flash[:error] = e.message
+          render_report_form
+        end
+
+        private
+
+        def respond_to_invalid_parameters
+          flash[:error] = I18n.t("order_management.reports.enterprise_fee_summary.invalid_filter_parameters")
+          render_report_form
+        end
+
+        def render_report_form
+          render action: :index
+        end
+
+        def render_report
+          return render_html_report unless @report.renderer.independent_file?
+
+          send_data(@report.render, filename: @report.filename)
+        end
+
+        def render_html_report
+          render action: :index
+        end
+
+        def report_renderer_klass
+          case params[:report_format]
+          when "csv"
+            OrderManagement::Reports::EnterpriseFeeSummary::Renderers::CsvRenderer
+          when nil, "", "html"
+            OrderManagement::Reports::EnterpriseFeeSummary::Renderers::HtmlRenderer
+          else
+            raise OpenFoodNetwork::Reports::UnsupportedReportFormatException
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
I actually moved the `lib/order_management/` files into a new engine `order_management`. (See https://github.com/openfoodfoundation/openfoodnetwork/pull/3115/commits/789b30f6faa62e85a0e48799792e09fe77509212 and https://github.com/openfoodfoundation/openfoodnetwork/pull/3115/commits/4f23685ea87341451803c98ca594adf19a6b2655.) But the controller would be more complicated to move :worried: , so I left it in the main application for now. I think we should do that in another PR, as it might need a lot more discussion not related to the Enterprise Fee Summary.

Exposing endpoints from the engine I think would not be tricky after we've packaged core controller behaviours.

As for this file, I have refactored (also because Rubocop demands I do!) and resorted to defining [`report_klass`](https://github.com/kristinalim/openfoodnetwork/blob/feature-enterprise_fees_report/app/controllers/spree/admin/reports/enterprise_fee_summary_report_controller.rb#L45). I think this succeeds in making the class easier to read.